mateo-mustapic/assignments/liferay-app/sdk/portlets/todo-portlet/docroot/js/main.js

```diff
@@ -31,21 +77,64 @@ AUI.add(
 									var listItem = currentTarget.ancestor('li');
 
 									listItem.remove();
+
+									if (finishedTask) {
+
+										var finishedHtml = A.Lang.sub(
+											TPL_FINISHED_TASK,
+											{
+												taskFinished: instance._taskFinished
+											}
+										);
+
+										finishedTask.append(finishedHtml);
+
+										remainingCount();
+
+										var taskHistory = A.one('.history-list');
+
+										var emptyTaskHistory = A.one('.empty-task-history');
```

Comment:
You have global DOM searches on each click action. You should:
1. Not have global DOM searches.
1. If you must have global DOM searched, have them once in initializer() and reference them as ```instance._myNode```